dask/bag/core.py

```diff
@@ -980,15 +988,38 @@ def foldby(self, key, binop, initial=no_default, combine=None,
         def combine2(acc, x):
             return combine(acc, x[1])
 
+        depth = 0
+        k = self.npartitions
+        b = a
+        while k > split_every:
+            c = b + str(depth)
+            if combine_initial is not no_default:
+                dsk2 = dict(((c, i), (reduceby, 0, combine2,
+                                             (toolz.concat, (map, dictitems,
+                                             [(b, j) for j in inds])),
+                                             combine_initial))
+                            for i, inds in enumerate(partition_all(split_every,
+                                                                   range(k))))
+            else:
+                dsk2 = dict(((c, i), (merge_with, (partial, reduce, combine),
+                                      [(b, j) for j in inds]))
+                           for i, inds in enumerate(partition_all(split_every, range(k))))
```

Comment:
We used to use the `dict((k, v) for ...)` syntax to support Python 2.6.  We have since dropped this.  Feel free to use the `{k: v for ...}` if you like.  It's also marginally faster.